### PR TITLE
Add Ruby gems workflow

### PR DESCRIPTION
**Pull request by struniu**
and update specs

### CODE REVIEW

File: .github/workflows/publish_to_artifactory.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Get Artifactory credentials
+      env:
+        ARTIFACTORY_API_KEY: ${{ secrets.ARTIFACTORY_API_KEY }}
+        ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+      run: |
+        curl https://zdrepo.jfrog.io/artifactory/api/gems/gems-local/api/v1/api_key.yaml -u $ARTIFACTORY_USERNAME:$ARTIFACTORY_API_KEY > $HOME/.gem/credentials
+        chmod 0600 $HOME/.gem/credentials
+    - name: Build gem
+      run: |
+        gem build *.gemspec
```
Comment (mms-pl) [inline .github/workflows/publish_to_artifactory.yml]: Why stray from established patterns and remove the general gem name from this? Seems like a no-cost and adds extra security.

**Review by mms-pl — APPROVED** [review]

**Review by mms-pl — CHANGES_REQUESTED** [review]
This is a public gem, it should not be in artifactory

**Comment by struniu** [comment]
> This is a public gem, it should not be in artifactory

indeed! I've added Ruby gems workflow then

**Review by mms-pl — APPROVED** [review]
Outstanding move!

Do we use such workflow anywhere else? :)

File: .github/workflows/gempush.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Ruby Gem
+
+on:
+  push:
```
Comment (mms-pl) [inline .github/workflows/gempush.yml:4]: maybe on new release?

**Comment by struniu** [comment]
> Outstanding move!
> 
> Do we use such workflow anywhere else? :)

Yes, in most of open source stuff we have. From sell input-sanitizer

File: .github/workflows/gempush.yml (hunk 2 above)
Comment (struniu) [reply .github/workflows/gempush.yml:4]: I think release is a subset of tag flow.

**Pull request merged**